### DISCUSSION TITLE
How to change a default (e.g. `vim.diagnostic.virtual_text`)  in `lua/vim/config/settings.lua` from my `~/.config/lvim/config.lua`

### DISCUSSION THREAD

**Discussion by eulersson** [LunarVim/LunarVim#4549] [Q&A] [answered]
Hello, I am quite new to NeoVim LUA's system. My question was how to be able to drill a different default, in this case `vim.diagnostic.virtual_text`:

https://github.com/LunarVim/LunarVim/blob/9ee3b7b8846d7ed2fa79f03d67083f8b95c897f2/lua/lvim/config/settings.lua#L87

If I throw `vim.diagnostic.virtual_text = false` in my user file `~/.config/lvim/config.lua` it gets overwritten at some point after those default settings are set by LunarVim, I would like to hook that value in the right time.

I cannot see a setup function for being able to merge the defaults table with my own customization.

I also had a look at the docs and could not find what I was looking for https://www.lunarvim.org/docs/configuration

Thank you!

**Comment by juicy-g** [comment]
Try `vim.diagnostic.config({ virtual_text = false })`.

**Reply by eulersson** [reply]
It does not work.

However if I run it an a result of a key shortcut it works after pressing the key combination.

```lua
vim.diagnostic.config({ virtual_text = false }) <---- This at the root of ~/.config/lvim/config.lua gets ignored
lvim.builtin.which_key.mappings["lx"] = {
  function()
    if vim.diagnostic.config().virtual_text then
      vim.diagnostic.config({
        virtual_text = false,
        signs = true,
        underline = true,
      })
    else
      vim.diagnostic.config({
        virtual_text = true,
      })
    end
  end,
  "Toggle Inline Diagnostics"
}

```

**Reply by eulersson** [reply]
I guess it might be because of the defaults being set after I set it (line 101 in the following snippet):

https://github.com/LunarVim/LunarVim/blob/9ee3b7b8846d7ed2fa79f03d67083f8b95c897f2/lua/lvim/config/settings.lua#L77-L101

**Reply by eulersson** [reply]
OK, if I comment out all the plugins it is honoured. It must be one of the pluggins then! I will investigate... This my config.lua:

https://github.com/eulersson/dotfiles/blob/main/.config/lvim/config.lua

**Reply by eulersson** [reply]
distant.nvim was setting it!

**Comment by eulersson** [comment, marked as answer]
It's because a plugin I had was setting it! 

https://github.com/chipsenkbeil/distant.nvim/blob/de7288b1af6fdb2d2e7a8aa00f07a236261c2491/lua/distant-core/ui/window.lua#L314-L322
